app/services/auth_service_base.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+class AuthServiceBase
+  include HTTParty
+
+  class << self
+    def find_user_by_token(access_token)
+      decoded_token = decode_token(access_token)
```

Comment:
It would be nice if must-implement functions like `decode_token` and `validate_token!` and `extract_user_atts` were stubbed out here (& throw an error)

Reply:
I agreed, but the problem is test coverage. Seems really weird to be testing that it throws an error, but I can do it I guess.